### PR TITLE
DOC: RT03 fix for read_sql method

### PR DESCRIPTION
**Pull request by YashpalAhlawat**
- [x] xref #57416  (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/io/sql.py [hunk 1]
```diff
@@ -608,6 +608,8 @@ def read_sql(
     Returns
     -------
     DataFrame or Iterator[DataFrame]
+        Returns a single DataFrame if applicable, or an Iterator
+        containing multiple DataFrames.
```
Comment (datapythonista) [inline pandas/io/sql.py:612]: The type is already clear from the previous line, can you explain in the description what is being returned, not expanding on the type.

File: pandas/io/sql.py [hunk 2]
```diff
@@ -625,7 +627,7 @@ def read_sql(
     Examples
     --------
     Read data from SQL via either a SQL query or a SQL tablename.
-    When using a SQLite database only SQL queries are accepted,
+    When using a SQLite database, only SQL queries are accepted,
```
Comment (datapythonista) [inline pandas/io/sql.py:630]: No big deal, but personally I think it's better to not have the comma.

**Review by datapythonista — COMMENTED** [review]
Thanks for the contribution @YashpalAhlawat, I added a couple of comments.

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @YashpalAhlawat

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @YashpalAhlawat

**Pull request merged**